orte/mca/odls/default/odls_default_module.c

```diff
@@ -400,6 +403,41 @@ static void send_error_show_help(int fd, int exit_status,
     exit(exit_status);
 }
 
+/* close all open file descriptors w/ exception of stdin/stdout/stderr,
+   the pipe used for the IOF INTERNAL messages, and the pipe up to
+   the parent. */
+static int close_open_file_descriptors(int write_fd,
+                                      orte_iof_base_io_conf_t opts) {
+    int pid = getpid();
+    char *fds_dir = NULL;
+    int rc = asprintf(&fds_dir, "/proc/%d/fd", pid);
+    if (rc < 0) {
+        return ORTE_ERR_OUT_OF_RESOURCE;
+    }
+    DIR *dir = opendir(fds_dir);
+    free(fds_dir);
+    if (NULL == dir) {
+        return ORTE_ERR_FILE_OPEN_FAILURE;
+    }
+    struct dirent *files;
+    while (NULL != (files = readdir(dir))) {
+        if (0 == strncmp(files->d_name,".",1) ||
+            0 == strncmp(files->d_name,"..",2)) {
```

Comment:
These two string comparisons are technically not correct.

A file named ".foo" will match them as well.  And ".." will match the first test, making the second test kinda meaningless.

I think the intent of this loop is to find only the files in the directory that are numbers.  Indeed, in /proc/pid/fd, that should be the only types of files that are in that dir.  But if we only want the numeric files, perhaps using isdigit() on the first character would be a better test...?